definition/json.go

```diff
@@ -0,0 +1,85 @@
+package definition
+
+import (
+	"unsafe"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/modern-go/reflect2"
+	amcfg "github.com/prometheus/alertmanager/config"
+	commoncfg "github.com/prometheus/common/config"
+)
+
+// secretEncoder encodes Secret to plain text JSON,
+// avoding the default masking behavior of the structure.
+type secretEncoder struct{}
+
+func (encoder *secretEncoder) Encode(ptr unsafe.Pointer, stream *jsoniter.Stream) {
+	stream.WriteString(getStr(ptr))
+}
+
+func (encoder *secretEncoder) IsEmpty(ptr unsafe.Pointer) bool {
+	return len(getStr(ptr)) == 0
+}
+
+func getStr(ptr unsafe.Pointer) string {
+	return *(*string)(ptr)
+}
+
+// secretEncoder encodes SecretURL to plain text JSON,
+// avoding the default masking behavior of the structure.
+type secretURLEncoder struct{}
+
+func (encoder *secretURLEncoder) Encode(ptr unsafe.Pointer, stream *jsoniter.Stream) {
+	url := getURL(ptr)
+	if url.URL != nil {
+		stream.WriteString(url.String())
+	} else {
+		stream.WriteNil()
+	}
```

Comment:
Nit: Use the `IsEmpty` function below?
```suggestion
	if !encoder.IsEmpty(ptr) {
		stream.WriteString(url.String())
	} else {
		stream.WriteNil()
	}
```

Reply:
It still needs the `url`, to write the string